ui/lib/kubernetes/addon/decorators/fetch-config.js

```diff
@@ -0,0 +1,49 @@
+import Route from '@ember/routing/route';
+
+/**
+ * the overview, configure, configuration and roles routes all need to be aware of the config for the engine
+ * if the user has not configured they are prompted to do so in each of the routes
+ * decorate the necessary routes to perform the check in the beforeModel hook since that may change what is returned for the model
+ */
+
+export function withConfig() {
```

Comment:
I realized that I was commenting on PR's that I didn't love extending routes and forgot that's what I did here for the config. A decorator achieves the same thing in a nicer way I think.

Reply:
This is awesome and something new to me! So, essentially, you changed the fetchConfig route to a decorator instead. When are instances where we use a decorator like this versus creating a route and extending it?

Reply:
I think in this case going with a route to begin with wasn't quite right since there was no actual `FetchConfig` route defined. A route file should realistically correspond to a route defined in the router. 

This pattern would also work well in a case where a parent route might be extended by a child. By using the decorator on both classes instead you can share what is needed and not worry about overriding a hook on what would have been the parent class.

Reply:
Thanks for explaining and this is good to know! 🙃

Reply:
Really cool idea making a decorator! This pattern of checking for some sort of configuration seems to be more and more common. It'd be cool if there was a way to make this more extensible (maybe by passing in the ember data model that will be passed to `queryRecord`?) I think we could easily leverage this decorator in the pki engine, we'd just need to pass the `pki/issuer` model instead. Not suggesting for this PR, but just to chew on 🤔

Reply:
Ya if this emerges as a reoccurring pattern, not necessarily for config but something you need to fetch and then prompt the user with a call to action we could add some args to the decorator, tweak the property names and make it reusable!